crates/payload/basic/src/lib.rs

```diff
@@ -173,6 +173,39 @@ where
         let until = tokio::time::Instant::now() + self.config.deadline;
         let deadline = Box::pin(tokio::time::sleep_until(until));
 
+        #[cfg(feature = "optimism")]
+        if config.attributes.no_tx_pool {
```

Comment:
This is not what we'll need in the end, but should be around the right place. Will clean up impl once we get past the payload / execution errors.